multiversx/mx-chain-go
# Propagate sync missing snapshot nodes error

## Reasoning behind the pull request
- When a missing trie node is found during snapshot, the sync process is triggered for that node. If the sync fails, the trie db for that epoch should not be marked as complete.
  
## Proposed changes
- If the sync process that was triggered by a missing snapshot node fails, propagate the sync err. This will not mark the db as complete, because the sync failed.

## Testing procedure
- Normal testing procedure. 
- If everything is ok, take the dbs, remove a trie node from the db, and start a testnet with the faulty dbs. When the first snapshot occurs,  the sync process should be triggered, but the node is missing from all the nodes. This will result in the failing of the sync. Check that the db is not marked as active.

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
